app/controllers/houseConsignment/index/items/additionalReference/RemoveAdditionalReferenceYesNoController.scala

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.houseConsignment.index.items.additionalReference
+
+import controllers.actions._
+import forms.YesNoFormProvider
+import models.{ArrivalId, Index, Mode, UserAnswers}
+import pages.houseConsignment.index.items.additionalReference.{AdditionalReferenceNumberPage, AdditionalReferencePage, RemoveAdditionalReferenceNumberYesNoPage}
+import pages.sections.houseConsignment.index.items.additionalReference.AdditionalReferenceSection
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents}
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.houseConsignment.index.items.additionalReference.RemoveAdditionalReferenceYesNoView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class RemoveAdditionalReferenceYesNoController @Inject() (
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  actions: Actions,
+  getMandatoryPage: SpecificDataRequiredActionProvider,
+  formProvider: YesNoFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: RemoveAdditionalReferenceYesNoView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  private val form = formProvider("houseConsignment.index.items.additionalReference.removeAdditionalReferenceYesNo")
+
+  private def addAnother(arrivalId: ArrivalId, mode: Mode): Call =
+    Call("GET", "#")
+  // TODO: update with AddAnotherAdditionalReferenceController
+
+  def insetText(userAnswers: UserAnswers, houseConsignmentIndex: Index, itemIndex: Index, additionalReferenceIndex: Index): Option[String] = {
```

Comment:
Does this need to be an Option if we always return a `Some(additionalReferenceType + insetTestAddRef).

Reply:
@joecefnhedog  yeah, it return none when there's no additionalReferenceType- updated the code